### ISSUE TITLE
Circular structure errors

### ISSUE THREAD

**Issue by aequasi** [sventschui/next-plugin-preact#10]
So, just gave this a try. I'm getting "Circular structure in "getInitialProps" result of page "X"" errors now, and I don't even have "getInitialProps".

Commenting out `getStaticProps` removes the error, but there is nothing circular in there

**Comment by hi019** [comment]
Related? https://github.com/vercel/next.js/issues/17854

**Comment by aequasi** [comment]
Potentially, yeah

**Comment by sventschui** [comment]
That was a tricky one. So the main issue was that we did not include `preact/debug` on the server, what caused the `__self` property not being removed from props.

Loading `preact/debug` uncovered that `preact-render-to-string` is missing a call to `options_diff` (see https://github.com/preactjs/preact-render-to-string/pull/172).

Until preact-render-to-string is released in a new version I added a monkey-patch to next-plugin-preact.

The error should be gone when upgrading to `3.0.3`. Let me know if you experience any further issues.
